src/server/terrain/biomes.zig

```diff
@@ -142,16 +142,28 @@ const Stripe = struct { // MARK: Stripe
 fn hashGeneric(input: anytype) u64 {
 	const T = @TypeOf(input);
 	return switch(@typeInfo(T)) {
-		.bool => @intFromBool(input),
-		.@"enum" => @intFromEnum(input),
-		.int, .float => @as(std.meta.Int(.unsigned, @bitSizeOf(T)), @bitCast(input)),
+		.bool => hashCombine(hashInt(@intFromBool(input)), 0xbf58476d1ce4e5b9),
+		.@"enum" => hashCombine(hashInt(@as(u64, @intFromEnum(input))), 0x94d049bb133111eb),
+		.int, .float => blk: {
+			const value = @as(std.meta.Int(.unsigned, @bitSizeOf(T)), @bitCast(input));
+			break :blk hashInt(@as(u64, value));
+		},
+		.@"union" => blk: {
+			if(@hasDecl(T, "getHash")) {
+				break :blk input.getHash();
+			}
+			@compileError("Unsupported type  " ++ @typeName(T));
```

Comment:
Should this maybe just hash the active field and its tag?
You can use a switch with an inline else to get access to the tag name at comptime.

Reply:
We literally have no way of testing it, even if we implement it. That union case is there only because I happened to implement it for purpose of hashing ZonElement, but since we no longer do that, it's dead code. I would prefer not to invest time into code that is not run. Last time we did code before we actually used it we got `writeFloat` with return type of `T` instead of `void` and zig didn't even notice that we are not returning anything there, it is getting fixed with first use in #1189

Reply:
Then I would honestly prefer not having the union case at all for now.

Reply:
Good. Removed.